File: 1217-relative-sort-array/1217-relative-sort-array.java
class Solution {
    public int[] relativeSortArray(int[] arr1, int[] arr2) {
        List<Integer> result = new ArrayList<>();
        
        for (int i = 0; i < arr2.length; i++) {
            for (int j = 0; j < arr1.length; j++) {
                if (arr2[i] == arr1[j]) {
                    result.add(arr1[j]);
                    arr1[j] = -1;
                }
            }
        }


        List<Integer> remain = new ArrayList<>();

        for(int i = 0; i < arr1.length; i++) {
            if (arr1[i] > -1) {
                remain.add(arr1[i]);
            }
        }

        Collections.sort(remain);
        result.addAll(remain);

        int[] finalresult = new int[result.size()];

        for (int i = 0; i < result.size(); i++) {
            finalresult[i] = result.get(i);
        }

        return finalresult;

    }
}

File: 1217-relative-sort-array/1217-relative-sort-array.py
class Solution:
    def relativeSortArray(self, arr1: List[int], arr2: List[int]) -> List[int]:
        result = []

        for i in range(len(arr2)):
            for j in range(len(arr1)):
                if arr2[i] == arr1[j]:
                    result.append(arr1[j])
                    arr1[j] = -1
        
        remain = [x for x in arr1 if x != -1]
        
        remain.sort()
        
        result.extend(remain)
        
        return result
        
        